src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -86,15 +86,30 @@ Transactor::Transactor(
 
 void Transactor::calculateFee ()
 {
-    mFeeDue = STAmount (getApp().getFeeTrack().scaleFeeLoad(
+    LoadFeeTrack& lft = getApp().getFeeTrack();
+
+    mFeeDue = STAmount (lft.scaleFeeLoad(
         calculateBaseFee(), view().fees().base,
-            view().fees().units, view().flags() & tapADMIN));
+            lft.getLoadBase(), view().flags() & tapADMIN));
 }
```

Comment:
Here's a detailed justification for using `lft.getCodeBase()` instead of `view().fees().units` in the call to `lft.scaleFeeLoad()`.  I hope it makes sense.

It starts with a description of my changes to `Transactor::calculateBaseFee()`.  There was already a comment in that method that said "Returns the fee in fee units", which was not correct -- it was returning drops.  Also the declaration of that method in `Transactor.h` said "Should be in fee units.  FIXME".  Given these comments, I changed `Transactor::calculateBaseFee()` to return fee units.  `calculateBaseFee()` now returns an integer multiple (from 1 to n) of the value returned by `lft.getLoadBase()` (https://github.com/ripple/rippled/blob/develop/src/ripple/core/LoadFeeTrack.h#L88).  That returned value is the normal load factor in fee units (`256`) (https://github.com/ripple/rippled/blob/develop/src/ripple/core/LoadFeeTrack.h#L122).

With that change `calculateBaseFee()`, which used to return drops, now returns fee units.

Now we'll return to the call to `lft.scaleFeeLoad()`.

An important part of `lft.scaleFeeLoad()` is cross multiplication: https://github.com/ripple/rippled/blob/develop/src/ripple/core/impl/LoadFeeTrack.cpp#L92.  I want `lft.scaleFeeLoad()` to return drops.  So if the incoming `fee` changed from drops to fee units, I need to also change what I divide by from drops to fee units.  Since I'm dividing by the `referenceFeeUnits` argument, I changed that from drops (`view().fees().units`) to fee units (`lft.getLoadBase()`).  Since the units are the same on both sides of the fraction, those two numbers combined make a simple ratio.

I still get drops back from `lft.scaleFeeLoad()` because the middle argument, `view().fees().base`, is in drops.

Feel free to ask more questions.  This explanation is pretty dense.  I'm really not fond of fees having their very own units (although I understand the justification).  And there may be a better way to accomplish what I'm doing.